tests/common/mod.rs

```diff
@@ -18,3 +26,24 @@ pub async fn create_up_transport_mqtt<S: Into<String>>(
 
     Mqtt5Transport::new(TransportMode::InVehicle, config, authority_name.into()).await
 }
+/// starts a mosquitto docker container and returns the container and the host port
+pub async fn start_mosquitto() -> (ContainerAsync<GenericImage>, u16) {
```

Comment:
IMHO we should return `Result<ContainerAsync<GenericImage>, std::err::Error>` and leave it to the client code to retrieve the port and/or fail the test if necessary.

Reply:
true, then let's keep it the way it currently is